File: 0059-spiral-matrix-ii/0059-spiral-matrix-ii.py
class Solution(object):
    def generateMatrix(self, n):
        """
        :type n: int
        :rtype: List[List[int]]
        """
        count=1
        matrix = [[0] * n for _ in range(n)]
        for layer in range((n + 1) // 2):
            for left in range(layer,n-layer):
                matrix[layer][left]=count
                count+=1
            for top in range(layer+1,n-layer):
                matrix[top][n-layer-1]=count
                count=count+1   
            for right in range(n-layer-2,layer-1,-1):
                matrix[n-layer-1][right]=count
                count=count+1
            for bottom in range(n-layer-2,layer,-1):
                matrix[bottom][layer]=count
                count=count+1   

        return matrix            
       

        

File: 0059-spiral-matrix-ii/0059-spiral-matrix-ii.cpp
class Solution {
public:
    vector<vector<int>> generateMatrix(int n) { 
        vector<vector<int>>matrix(n,vector<int>(n,0));
        int count=1;
        for(int layer=0 ;layer<=n+1/2;layer++){
            for(int left=layer;left<n-layer;left++){
                matrix[layer][left]=count++;
            }
            for(int top=layer+1;top<n-layer;top++){
                matrix[top][n-layer-1]=count++;
            }
            for(int right=n-layer-2;right>=layer;right--){
                matrix[n-layer-1][right]=count++;
            }
            for(int bottom=n-layer-2;bottom>=layer+1;bottom--){
                matrix[bottom][layer]=count++;
            }
        }
        return matrix;
        
    }
};